apps/src/templates/curriculumCatalog/CurriculumCatalogCard.jsx

```diff
@@ -91,65 +102,84 @@ const CustomizableCurriculumCatalogCard = ({
-        <FontAwesome icon="clock" className="fa-solid" />
-        <p>{duration}</p>
-      </div>
+  sectionsForDropdown,
+  ...props
+}) => {
+  const [isAssignDialogOpen, setIsAssignDialogOpen] = useState(false);
+
+  return (
+    <>
       <div
         className={classNames(
-          style.buttonsContainer,
+          style.curriculumCatalogCardContainer,
           isEnglish
-            ? style.buttonsContainer_english
-            : style.buttonsContainer_notEnglish
+            ? style.curriculumCatalogCardContainer_english
+            : style.curriculumCatalogCardContainer_notEnglish
         )}
       >
-        <Button
-          __useDeprecatedTag
-          color={Button.ButtonColor.neutralDark}
-          type="button"
-          href={pathToCourse}
-          aria-label={quickViewButtonDescription}
-          text={quickViewButtonText}
-        />
-        <Button
-          color={Button.ButtonColor.brandSecondaryDefault}
-          type="button"
-          onClick={() => {}}
-          aria-label={assignButtonDescription}
-          text={assignButtonText}
-        />
+        <img src={imageSrc} alt={imageAltText} />
+        <div className={style.curriculumInfoContainer}>
+          <div className={style.labelsAndTranslatabilityContainer}>
+            <div className={style.labelsContainer}>
+              <CardLabels subjectsAndTopics={subjectsAndTopics} />
+            </div>
+          </div>
+          {isTranslated && (
+            <FontAwesome
+              icon="language"
+              className="fa-solid"
+              title={translationIconTitle}
+            />
+          )}
+          <h4>{courseDisplayName}</h4>
+          <div className={style.iconWithDescription}>
+            <FontAwesome icon="user" className="fa-solid" />
+            <p>{gradeRange}</p>
+          </div>
+          <div className={style.iconWithDescription}>
+            <FontAwesome icon="clock" className="fa-solid" />
+            <p>{duration}</p>
+          </div>
+          <div
+            className={classNames(
+              style.buttonsContainer,
+              isEnglish
+                ? style.buttonsContainer_english
+                : style.buttonsContainer_notEnglish
+            )}
+          >
+            <Button
+              __useDeprecatedTag
+              color={Button.ButtonColor.neutralDark}
+              type="button"
+              href={pathToCourse}
+              aria-label={quickViewButtonDescription}
+              text={quickViewButtonText}
+            />
+            <Button
+              color={Button.ButtonColor.brandSecondaryDefault}
+              type="button"
+              onClick={() => {
+                setIsAssignDialogOpen(true);
+              }}
+              aria-label={assignButtonDescription}
+              text={assignButtonText}
+            />
+          </div>
+        </div>
       </div>
-    </div>
-  </div>
-);
+      {isAssignDialogOpen && (
```

Comment:
This part, the `const [isAssignDialogOpen, setIsAssignDialogOpen] = useState(false);` part above, and the spreading of props is the only thing that actually changed in this component 🤯